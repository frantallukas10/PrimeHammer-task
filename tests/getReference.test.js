const getReference = require('../helpers/getReference')

describe('Test check getReference function', () => {
  let check, thermometer, humidity
  test('1st sample', () => {
    check = 'reference'
    thermometer = '70.0'
    humidity = '45.0'
    expect(getReference(check, thermometer, humidity)).toMatchSnapshot()
  })

  test('2st sample', () => {
    check = ''
    thermometer = '70.0'
    humidity = '45.0'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('3st sample', () => {
    check = 'reference'
    thermometer = ''
    humidity = ''
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('4st sample', () => {
    check = 'reference'
    thermometer = ''
    humidity = '45.0'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('5st sample', () => {
    check = 'reference'
    thermometer = 'bla'
    humidity = 'bla'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('6st sample', () => {
    check = 'reference'
    thermometer = 'undefined'
    humidity = '10'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('7st sample', () => {
    check = 'reference'
    thermometer = 'null'
    humidity = '10'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
  test('8st sample', () => {
    check = 'reference'
    thermometer = 'NaN'
    humidity = '10'
    expect(() => {
      getReference(check, thermometer, humidity)
    }).toThrowErrorMatchingSnapshot()
  })
})
