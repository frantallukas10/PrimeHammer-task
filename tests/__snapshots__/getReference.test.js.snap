// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test check getReference function 1st sample 1`] = `
Object {
  "humidity": 45,
  "thermometer": 70,
}
`;

exports[`Test check getReference function 2st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: 45.0"
`;

exports[`Test check getReference function 3st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: "
`;

exports[`Test check getReference function 4st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: 45.0"
`;

exports[`Test check getReference function 5st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: bla"
`;

exports[`Test check getReference function 6st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: 10"
`;

exports[`Test check getReference function 7st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: 10"
`;

exports[`Test check getReference function 8st sample 1`] = `
"First line of file should start with reference line and
      it shoud contain referenc text,
      thermometer value should number,
      humidity value should number.
      actual values are: 10"
`;
